src/dtx/dtx_common.c

```diff
@@ -0,0 +1,528 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * dtx: DTX common logic
+ */
+#define D_LOGFAC	DD_FAC(dtx)
+
+#include <uuid/uuid.h>
+#include <abt.h>
+#include <daos_srv/pool.h>
+#include <daos_srv/container.h>
+#include <daos_srv/vos.h>
+#include <daos_srv/dtx_srv.h>
+#include <daos_srv/daos_server.h>
+#include "dtx_internal.h"
+
+struct dtx_batched_commit_args {
+	d_list_t		 dbca_link;
+	struct ds_cont_hdl	*dbca_cont_hdl;
+};
+
+void
+dtx_aggregate(void *arg)
+{
+	struct ds_cont		*cont = arg;
+
```

Comment:
I think it would be better to set dtx_aggregating here.  Is there a benefit to setting the flag outside?   Otherwise, you are blocking the thread that needs to close the handle where the aggregation thread that has yet to execute will start, and exit anyway and unset the flag anyway.

Unless there is a good reason to do otherwise, it would keep the modification of that flag internal to this routine.

Reply:
Ok, that is reasonable.   It's mostly another asymmetry issue where we set a flag at higher level and read at lower level.   Perhaps some is_set/set/clear macros would suffice for that issue.   I'm not requesting changes on this patch.